lisa/sut_orchestrator/azure/arm_template.bicep

```diff
@@ -0,0 +1,349 @@
+
+resource virtual_network_name_resource 'Microsoft.Network/virtualNetworks@2020-05-01' = if (empty(virtual_network_resource_group)) {
+  name: virtual_network_name
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        '10.0.0.0/16'
+      ]
+    }
+    subnets: [for j in range(0, subnet_count): {
+      name: '${subnet_prefix}${j}'
+      properties: {
+        addressPrefix: '10.0.${j}.0/24'
+      }
+    }]
+  }
+}
+
+resource availability_set 'Microsoft.Compute/availabilitySets@2019-07-01' = if (use_availability_set) {
+  name: availability_set_name_value
+  location: location
+  tags: availability_set_tags
+  sku: {
+    name: 'Aligned'
+  }
+  properties: availability_set_properties
+}
+
+resource nodes_public_ip 'Microsoft.Network/publicIPAddresses@2020-05-01' = [for i in range(0, node_count): {
+  location: location
+  name: '${nodes[i].name}-public-ip'
+  properties: {
+    publicIPAllocationMethod: (is_ultradisk ? 'Static' : 'Dynamic')
+  }
+  sku: {
+    name: (is_ultradisk ? 'Standard' : 'Basic')
+  }
+  zones: (use_availability_zones ? availability_zones : null)
+}]
+
+module nodes_nics './nested_nodes_nics.bicep' = [for i in range(0, node_count): {
+  name: '${nodes[i].name}-nics'
+  params: {
+    vmName: nodes[i].name
+    nic_count: nodes[i].nic_count
+    location: location
+    vnet_id: vnet_id
+    subnet_prefix: subnet_prefix
+    existing_subnet_ref: existing_subnet_ref
+    enable_sriov: nodes[i].enable_sriov
+  }
+}]
+
+resource nodes_image 'Microsoft.Compute/images@2019-03-01' = [for i in range(0, node_count): if (isVhd(nodes[i]) && empty(nodes[i].vhd.vmgs_path)) {
+  name: '${nodes[i].name}-image'
+  location: location
+  properties: {
+    storageProfile: {
+      osDisk: {
+        osType: 'Linux'
+        osState: 'Generalized'
+        blobUri: nodes[i].vhd.vhd_path
+        storageAccountType: 'Standard_LRS'
+      }
+    }
+    hyperVGeneration: 'V${nodes[i].hyperv_generation}'
+  }
+}]
+
+resource nodes_disk 'Microsoft.Compute/disks@2021-04-01' = [for i in range(0, node_count): if (isCvm(nodes[i])) {
+  name: '${nodes[i].name}-disk'
+  location: location
+  sku: {
+    name: 'Standard_LRS'
+  }
+  properties: {
+    osType: 'Linux'
+    hyperVGeneration: 'V${nodes[i].hyperv_generation}'
+    securityProfile: {
+      securityType: 'ConfidentialVM_VMGuestStateOnlyEncryptedWithPlatformKey'
+    }
+    creationData: {
+      createOption: 'ImportSecure'
+      storageAccountId: resourceId(shared_resource_group_name, 'Microsoft.Storage/storageAccounts', vhd_storage_name)
+      securityDataUri: nodes[i].vhd.vmgs_path
+      sourceUri: nodes[i].vhd.vhd_path
+    }
+  }
+  zones: (use_availability_zones ? availability_zones : null)
+}]
+
+resource nodes_data_disks 'Microsoft.Compute/disks@2022-03-02' = [for i in range(0, (length(data_disks) * node_count)): if (data_disks[(i % length(data_disks))].type == 'UltraSSD_LRS') {
```

Comment:
Move `if` to next line, so it's easy to read. Add comments to explain the different logic of UltraSSD_LRS. Can the logic of ultraSSD_LRS be in a function?

Reply:
no need the if condition during the testing. it can't be put under the function, since the resource will check the properties' names